test cases/common/43 has function/meson.build

```diff
@@ -88,4 +90,22 @@ foreach cc : compilers
     assert (cc.has_function('sendmmsg', args : unit_test_args),
             'Failed to detect function "sendmmsg" (should always exist).')
   endif
+
+  # On MinGW, strndup was incorrectly detected as available with has_function
+  # even though it is not, so if it is detected by has_function, assert that it
+  # can actually be found by the linker.
+  if cc.has_function('strndup')
+    assert(cc.has_function('strndup', prefix: '#include <string.h>'), 'strndup not found with prefix')
+    assert(cc.links('#include <string.h>\nint main() { char *s = strndup ("string", 1); }'), 'strndup incorrectly detected')
+  endif
+
+  # This is meant to be used as __builtin_smul_overflow, so we shouldn't find the unbuiltin version
+  if ['gcc', 'clang'].contains(cc.get_id())
+    assert(cc.has_function('__builtin_smul_overflow'), '__builtin_smul_overflow must be found')
+    # FIXME: this fails on gcc, and we can't fix it without breaking builtin
+    # detection without headers for things like alloca()
```

Comment:
I think with the implicit include workaround we can enable this test for gcc as well.